src/WorkOS.net/Services/Webhooks/WebhookService.cs

```diff
@@ -0,0 +1,219 @@
+        /// <param name="secret">The webhook endpoint's signing secret.</param>
+        /// <param name="tolerance">The time tolerance, in seconds.</param>
+        /// <returns>The deserialized JSON.</returns>
+        public Webhook ConstructEvent(
+            string json,
+            string signatureHeader,
+            string secret,
+            long tolerance)
+        {
+            if (this.VerifyHeader(json, signatureHeader, secret, tolerance))
+            {
+                Webhook webhookVerified = JsonConvert.DeserializeObject<Webhook>(json);
+                return webhookVerified;
+            }
+            else
+            {
+                throw new Exception("Unable to verify Webhook");
+            }
+        }
+
+        /// <summary>
+        /// Verify Header.
+        /// </summary>
+        /// <param name="payload">
+        /// json payload.
+        /// </param>
+        /// <param name="sigHeader">
+        /// signature header of webhook.
+        /// </param>
+        /// <param name="secret">
+        /// secret.
+        /// </param>
+        /// <param name="tolerance">
+        /// time tolerance for timing attacks.
+        /// </param>
+        /// <returns>Boolean True or False of verification.</returns>
+        public bool VerifyHeader(string payload, string sigHeader, string secret, long tolerance)
+        {
+            var timeAndSignature = this.GetTimestampAndSignature(sigHeader);
+            var timeStamp = timeAndSignature.Item1;
+
+            if (!this.Verify_Time_Tolerance(timeStamp, tolerance))
+            {
+                throw new Exception("Timestamp outside of the tolerance zone");
+            }
+
+            string signatureHash = timeAndSignature.Item2;
+            string expectedSig = this.ComputeSignature(timeStamp, payload, secret);
+
+            if (this.SecureCompare(expectedSig, signatureHash))
+            {
+                return true;
+            }
+            else
+            {
+                throw new Exception("Signature hash does not match the expected signature hash for payload");
+            }
+        }
+
+        /// <summary>
+        /// Get timestamp and signature hash.
+        /// </summary>
+        /// <param name="signatureHeader">
+        /// Signatures header.
+        /// </param>
+        /// <returns> Tuple of [DateTime timestamp, string signaturehash].</returns>
+        public (string Timestamp, string SignatureHash) GetTimestampAndSignature(string signatureHeader)
+        {
+            var timeAndSig = signatureHeader.Split(',');
+            var timeStamp = timeAndSig[0];
+            var signatureHash = timeAndSig[1];
+            if (string.IsNullOrEmpty(timeStamp) || string.IsNullOrEmpty(signatureHash))
+            {
+                throw new ArgumentException("Unable to extract timestamp and signature hash from header");
+            }
+
+            var removeT = "t=";
+            var t = timeStamp.IndexOf(removeT);
```

Comment:
I think this could be cleaned up a bit using `StartsWith`, since we know that the prefix is going to be at the start of the string:

```cs
const TimestampPrefix = "t=";

if (timeStamp.StartsWith(TimestampPrefix)
{
    timeStamp = timeStamp.Substring(TimestampPrefix.Length);
}
```